docs/src/content/docs/core/define-integration.mdx

```diff
@@ -15,14 +15,21 @@ import {
     defineOptions,
 } from "astro-integration-kit";
 import { corePlugins } from "astro-integration-kit/plugins";
+import { z } from "astro/zod";
 
-type Options = {
-    foo?: string | undefined;
+const OptionsSchema = z.object({
+    foo: z.string().optional().default("bar"),
+});
+
+export type Options = {
+    foo?: z.infer<typeof OptionsSchema>["foo"];
 }
 
 export default defineIntegration({
     name: "my-integration",
-    options: defineOptions<Options>({ foo: "bar" }),
+    options: defineOptions<Options>({
+        schema: OptionsSchema,
+    }),
```

Comment:
How does it behave (type-wise) with schemas with different input and output types?